tests/test_auth.py

```diff
@@ -111,3 +114,35 @@ async def test_auth_bad_credentials(
     call = calls[0]
     assert call[1]["data"] == AUTH_DATA
     assert call[1]["headers"] == AUTH_HEADERS
+
+
+async def test_user_details_requested_only_once(
+    auth_mock: Auth, backend_selector_mock: BackendSelector, aioresponses_mock
+):
+    aioresponses_mock.get(
+        backend_selector_mock.user_details_url, payload={"accountId": ACCOUNT_ID}
+    )
+
+    headers = {
+        "Authorization": f"Bearer {auth_mock.get_access_token()}",
+        "Content-Type": "application/json",
+        "User-Agent": "okhttp/3.12.0",
+        "Pragma": "no-cache",
+        "Cache-Control": "no-cache",
+    }
+
+    await auth_mock.get_account_id()
+
+    aioresponses_mock.assert_called_with(
+        backend_selector_mock.user_details_url, "GET", headers=headers
+    )
+
+    assert auth_mock._auth_dict["accountId"] == ACCOUNT_ID
```

Comment:
```suggestion
```
you can reset `aioresponses_mock` and check that the call count for the URL is zero after the second call to `get_account_id()` instead of accessing the internal dict